pennylane/ops/arithmetic/sum.py

```diff
@@ -0,0 +1,144 @@
+    """Compute the sum of the provided terms"""
+    return Sum(*summands)  # a wire order is required when combining operators of varying sizes
+
+
+class Sum(Operator):
+    """Arithmetic operator subclass representing the sum of operators"""
+
+    def __init__(self, *summands, do_queue=True, id=None):
+
+        if len(summands) < 2:
+            raise ValueError(f"Require at least two operators to sum; got {len(summands)}")
+
+        self.summands = summands
+
+        combined_wires = qml.wires.Wires.all_wires([s.wires for s in summands])
+        combined_params = [s.parameters for s in summands]
+        super().__init__(
+            *combined_params, wires=combined_wires, do_queue=do_queue, id=id
+        )
+        self._name = "Sum"
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " + ".join([f"{f}" for f in self.summands])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+        copied_op.summands = tuple(s.__copy__() for s in self.summands)
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "summands"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @property
+    def is_hermitian(self):
+        """If all of the terms in the sum are hermitian, then the Sum is hermitian."""
+        return all([s.is_hermitian for s in self.summands])
+
+    def terms(self):
+        return [1.0]*len(self.summands), self.summands
+
+    def diagonalizing_gates(self):
+        """Compute diagonalizing_gates (only if op is hermitian)"""
+        if not self.is_hermitian:
+            raise qml.operation.DiagGatesUndefinedError
+
+        Hmat = self.matrix()
+        Hmat = qml.math.to_numpy(Hmat)
+        Hkey = tuple(Hmat.flatten().tolist())
+        eigs = {}
+        if Hkey not in eigs:
+            w, U = np.linalg.eigh(Hmat)
+            eigs[Hkey] = {"eigvec": U, "eigval": w}
+
+        eigenvectors = eigs[Hkey]["eigvec"]
+        return [qml.QubitUnitary(eigenvectors.conj().T, wires=self.wires)]
+
+    def matrix(self, wire_order=None):
+        """Representation of the operator as a matrix in the computational basis."""
+
+        def matrix_gen(summands, wire_order=None):
+            """Helper function to construct a generator of matrices"""
+            for op in summands:
+                yield expand_matrix(op.matrix(), op.wires, wire_order=wire_order)
+
+        if wire_order is None:
+            wire_order = self.wires
+
+        return self._sum(matrix_gen(self.summands, wire_order))
+
+    def eigvals(self, wire_order=None):
+        """Get eigvals of the the Sum of operators"""
+        if not self.is_hermitian:
+            raise qml.operation.EigvalsUndefinedError  # eigvals for non-hermitian obs can be imaginary
```

Comment:
Nothing wrong with imaginary eigenvalues.  We should still be able to return them. For example:

```
>>> qml.RX(1.2, wires=0).eigvals()
array([0.82533561+0.56464247j, 0.82533561-0.56464247j])
```

Also if `diagonalizing_gates` is going to use `eigh` and to compute both eigenvalues and diagonalizing gates, we could use cached eigenvalues instead of recomputing them.

Also, `eigvals` does not take a `wire_order` keyword.

Reply:
Cool, so we can allow for imaginary eigvals here, but just inhibit the use inside `measurementprocesses`